services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/DirectionsResponseTest.java

```diff
@@ -83,6 +85,15 @@ public void fromJson_correctlyBuildsFromJson() throws Exception {
     assertEquals(1, response.routes().size());
   }
 
+  @Test
+  public void deserialization_from_byte_buffer() throws Exception {
+    String json = loadJsonFixture(DIRECTIONS_V5_PRECISION6_FIXTURE);
+    ByteBuffer buffer = ByteBuffer.wrap(json.getBytes(StandardCharsets.UTF_8));
```

Comment:
I just want to point out that you can only pass `java.nio.DirectBuffer` (created with `java.nio.ByteBuffer.allocateDirect`, or the JNI `NewDirectByteBuffer` function) to native using the [DataRef type](https://github.com/mapbox/mapbox-bindgen/pull/1410). `ByteBuffer.wrap` doesn't allocate memory on heap so native can't access the memory region directly. Otherwise, code from `DirectionsResponse` looks good to me.

Reply:
Thank you for the clarification. I updated the test to use direct buffer to avoid confusion.